### PR TITLE
SONARKT-301 stop raising issues on abstract/overrided/open/interface methods

### PR DESCRIPTION
**Pull request by erwan-serandour-sonarsource**

### CODE REVIEW

File: kotlin-checks-test-sources/src/main/files/non-compiling/checks/CollectionShouldBeImmutableCheckSampleNonCompiling.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package checks
+
+class CollectionShouldBeImmutableCheckSampleNonCompiling {
+
+    actual fun actualFun(list: MutableList<Int>): Unit // compliant
+    expect fun expectFun(list: MutableList<Int>): Unit // compliant
+
+    fun qualifiedStarange() {
```
Comment (johann-beleites-sonarsource) [inline kotlin-checks-test-sources/src/main/files/non-compiling/checks/CollectionShouldBeImmutableCheckSampleNonCompiling.kt:8]:
```suggestion
    fun qualifiedStrange() {
```

File: sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/CollectionShouldBeImmutableCheck.kt [hunk 2]
```diff
@@ -52,8 +59,25 @@ class CollectionShouldBeImmutableCheck : AbstractCheck() {
             "kotlin.collections.MutableCollection"
         )
 
+    override fun visitClass(klass: KtClass, context: KotlinFileContext) {
+        if (klass.isInterface()) return
+
+        klass
+            .collectDescendantsOfType<KtNamedFunction>(::notAnInterface) { true }
+            .forEach { reportNamedFunction(it, context) }
+    }
```
Comment (johann-beleites-sonarsource) [inline sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/CollectionShouldBeImmutableCheck.kt:68]: Like this, we would miss any cases that are inside of file-level functions. Like this you can get those too:

```suggestion
    override fun visitKtFile(file: KtFile, context: KotlinFileContext) {
        file
            .collectDescendantsOfType<KtNamedFunction>(::notAnInterface) { true }
            .forEach { reportNamedFunction(it, context) }
    }
```

Comment (johann-beleites-sonarsource) [reply sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/CollectionShouldBeImmutableCheck.kt:68]: You can put this on file level into the sample file:
```
private fun nonCompliantParameterOnFileLevel(list: MutableList<Int>): Int { // Noncompliant
    return list.reduce { acc, it -> acc + it}
}
```

File: kotlin-checks-test-sources/src/main/files/non-compiling/checks/CollectionShouldBeImmutableCheckSampleNonCompiling.kt (hunk 1 above)
Comment (erwan-serandour-sonarsource) [reply kotlin-checks-test-sources/src/main/files/non-compiling/checks/CollectionShouldBeImmutableCheckSampleNonCompiling.kt:8]: :+1:

**Review by johann-beleites-sonarsource — APPROVED** [review]

**Pull request merged**